rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystemStatsProvider.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.core;
+
+import org.weakref.jmx.Managed;
+
+import static com.qubole.rubix.spi.utils.DataSizeUnits.BYTES;
+
+/**
+ * Created by stagra on 28/1/16.
+ */
+public class CachingFileSystemStatsProvider
+{
+  private ReadRequestChainStats stats;
+
+  public CachingFileSystemStatsProvider ()
+  {
+    this.stats = new ReadRequestChainStats();
+  }
+
+  public synchronized void addReadRequestChainStats(ReadRequestChainStats newStats)
+  {
+    this.stats = this.stats.add(newStats);
```

Comment:
When we use CachingFileSystemStatsProvider in Bookeeper as static member, will we end up accumulating ReadRequestChainStats always ?

Reply:
Did not get the question. The provider in BookKeeper will only accumulate stats for parallel warmup and non-local request, rest will be done by providers in CachingFileSystem

Reply:
If I understood the code correctly, we're accumulating stats in an object over the full lifetime of BookKeeper. Always increasing counters are not useful in monitoring dashboards and alerts. I think something like `CounterStat` from airlift might be suitable as that will give us moving averages over 1, 5, 15min and a cumulative counter.

Reply:
Ya,  Although these stats are meant to give overall picture (cummulative one) right now, moving averages could be useful. I will open an issue to add CounterStats over this.